### DISCUSSION TITLE
isFitted Epic

### DISCUSSION THREAD

**Discussion by andresin87** [SUI-Components/sui-components#1667] [Ideas and new features]
There is some different considerations implemented all over the code which tackles the styling rules about each component's margins and paddings and borders also. Sometimes, there are extra spacing, in other cases there is not, and some component, have spacings in weird situations.

The final decision is to deliver (publish) each component as it is (with No extra spacing) for all the cases.

For the transition, until we publish a major of any spaced component we are going to have a boolean prop which enables this desired behavior in those needed components. This prop will be removed in its next major published version.

The prop name must be agnostic to its technical implementation. We decided to name it `isFitted`. ~~isMarginless~~, ~~isPaddingless~~, ~~isBorderless~~

- Agreements Explanation:
- Tasks list:
	- [x] atom/switch:
	  - https://github.com/SUI-Components/sui-components/pull/1661
	  - https://github.com/SUI-Components/sui-components/pull/1663 @xaviermariaconejo 
	  - https://github.com/SUI-Components/sui-components/pull/1670 @oriolpuig @fredyzz 
    - [x] atom/button
    - [x] atom/tag
	  - https://github.com/SUI-Components/sui-components/pull/1853
    - [x] atom/badge
      - https://github.com/SUI-Components/sui-components/pull/1692 @adriadoz

**Comment by turolopezsanabria** [comment]
Also reported as an issue here:
https://github.com/SUI-Components/sui-components/issues/1708

**Comment by IsabelGomez87** [comment]
The modal component also need this feature `isFitted`
